src/app/project-sets/page.tsx

```diff
@@ -0,0 +1,51 @@
+'use client'
+
+import * as React from "react"
+import {Typography} from "@/components/ui/typography"
+import {columns, type ProjectSet} from "@/app/project-sets/columns"
+import { DataTable } from "@/components/ui/data-table"
+
+async function getData(): Promise<ProjectSet[]> {
+    // Fetch data from your API here.
+    return Array.from(Array(21300)).map((_, index) => {
+        const idx = index + 1
+        // give me a random integer
+        const numProjects = Math.floor(Math.random() * 100)
+        return {
+            id: index,
+            name: "Project Set " + idx,
+            numProjects: numProjects,
+        }
+    })
+}
+
+function ProjectsPage() {
+    const [projectSets, setProjectSets] = React.useState<ProjectSet[]>([])
+
+    React.useEffect(() => {
```

Comment:
I have to do this since I make this component 'use client' because they don't allow passing function as props from server to client component

Reply:
In general, I think making a page use client in it's entirety is almost never necessary in next14 and I would posit to largely avoid doing it specifically for this reason.

Pages should remain server components specifically because it way simplifies the data fetching needed at the top level like this.

The table can be its own component. Semantically, we want the page to render on the server, and the table to render on client. Then we can make the responsibility of page components to fetch any needed data for the page, and pass it to their children.

Reply:
Make a component use client should not be a trivial decision made to avoid errors, but a thoughtful decision about the responsibilities of the component and what can be delegated

Reply:
I found a wayyyyyy